lib/wcc/contentful/client_ext.rb

```diff
@@ -6,7 +6,7 @@ class << self
   end
 
   def self.get_http(url, query, headers = {}, proxy = {})
-    if override = WCC::Contentful.configuration.http_adapter
+    if override = WCC::Contentful::SimpleClient.load_adapter(WCC::Contentful.configuration.http_adapter)
```

Comment:
This is going to create the adapter for each HTTP get call.  Could we extract this into a memoized property?

```rb
def override
  @override ||= WCC::Contentful::SimpleClient.load_adapter(WCC::Contentful.configuration.http_adapter)
end

...
def self.get_http(url, query, headers = {}, proxy = {})
    if override
      override.call(url, query, headers, proxy)


```

Reply:
Actually, even that is going to be calling into load_adapter for each HTTP call if the adapter returns nil.  We could do this:

```rb
def adapter
  @adapter ||=
    WCC::Contentful::SimpleClient.load_adapter(WCC::Contentful.configuration.http_adapter) ||
    ->(url, query, headers, proxy) { old_get_http(url, query, headers, proxy) }
end

...
def self.get_http(url, query, headers = {}, proxy = {})
    adapter.call(url, query, headers, proxy)
```

We'd no longer need the if statement in `get_http`, because we are determining the adapter once, lazily, inside `def adapter`